client/dynamicplugins/registry.go

```diff
@@ -301,6 +303,46 @@ func (d *dynamicRegistry) ListPlugins(ptype string) []*PluginInfo {
 	return plugins
 }
 
+// WaitForPlugin repeatedly checks until a plugin with a given type and name
+// becomes available or its context is canceled or times out.
+// Callers should pass in a context with a sensible timeout
+// for the plugin they're expecting to find.
+func (d *dynamicRegistry) WaitForPlugin(ctx context.Context, ptype, name string) (*PluginInfo, error) {
+	// these numbers are almost arbitrary...
+	delay := 200     // milliseconds between checks, will backoff
+	maxDelay := 5000 // up to 5 seconds between each check
+
+	// put a long upper bound on total time,
+	// just in case callers don't follow directions.
+	ctx, cancel := context.WithTimeout(ctx, 24*time.Hour)
+	defer cancel()
+
+	timer := time.NewTimer(time.Duration(delay) * time.Millisecond)
+	for {
+		for _, p := range d.ListPlugins(ptype) {
+			if p.Name == name {
+				return p, nil
+			}
+		}
+
+		//log.Printf("WaitForPlugin delay: %d", delay) // TODO: get a logger in here?
```

Comment:
Debugging debris to clean up here.

Reply:
Sometimes I leave these on purpose for a later person debugging to simply uncomment, but sounds like we prefer not to do that, so will remove!